.github/workflows/publish.yml

```
@@ -0,0 +1,40 @@
+name: Publishing
+
+on:
+  # Only publish when 'All tests' has finished on master branch
+  workflow_run:
+    workflows: [ All tests ]
+    branches: [ master ]
+    types: [ completed ]
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  publish_snapshot:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    if: ${{ github.event.workflow_run.conclusion == 'success' }} # only publish if the tests passed
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 11
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
```

there are some new releases:

```suggestion
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-java@v4
        with:
          distribution: adopt
          java-version: 11

      - name: Setup Gradle
        uses: gradle/actions/setup-gradle@v4
```

no further changes required for these majors.